notifier/src/main/kotlin/modules/JiraNotifier.kt

```diff
@@ -47,6 +48,25 @@ class JiraNotifier(
         restClient.issueClient.addComment(issue.commentsUri, Comment.valueOf(comment)).claim()
     }
 
+    /**
+     * Change the [state] of the given issue specified by the [issueKey].
+     */
+    fun changeState(issueKey: String, state: String): Result<Void> {
+        val issue = restClient.issueClient.getIssue(issueKey).claim()
+        val possibleTransitions = restClient.issueClient.getTransitions(issue).claim()
+
+        val transition = possibleTransitions.singleOrNull { it.name == state } ?: return Result.failure(
+            IllegalArgumentException("The transition to the state '$state' is not possible.")
+        )
+
+        return runCatching {
+            restClient.issueClient.transition(issue, TransitionInput(transition.id)).claim()
+        }.onFailure {
+            log.error { "The transition to state '$state' for the issue '$issueKey' is not possible: " +
+                    it.collectMessagesAsString() }
+        }
+    }
```

Comment:
So phrase out my previous comment, how about:

    fun changeState(issueKey: String, state: String): Boolean {
        val issue = restClient.issueClient.getIssue(issueKey).claim()
        val possibleTransitions = restClient.issueClient.getTransitions(issue).claim()

        return runCatching {
            val transition = possibleTransitions.single { it.name == state }
            restClient.issueClient.transition(issue, TransitionInput(transition.id)).claim()
        }.onFailure {
            log.error { "The transition to state '$state' for the issue '$issueKey' is not possible: " +
                    it.collectMessagesAsString() }
        }.isSuccess
    }

Reply:
Wouldn't it be better to use `Result<>` for two reasons:
1. Consistency with the `createIssue()` method.
2. Explicitly stating what the return value of the method represents?

Reply:
> Wouldn't it be better to use `Result<>` for two reasons:

Personally, I don't have a clear opinion on that, and will leave it to you guys. I'll approve in any case.

> Explicitly stating what the return value of the method represents?

What's not explicit about the `Boolean` return value?

Reply:
> What's not explicit about the Boolean return value?

`Result<Void>` specifies that a http request is being made.
But I'm okay with either, and won't block.

Reply:
> `Result<Void>` specifies that a http request is being made.

Uhm, does it? How? Because you know by heart which functions return `Void`?

Reply:
I think we can leave it as it is for now.

Reply:
Yes, you're right. It's not http request related, but `Result` explicitly states that it can fail or succeed.
A Boolean return value only implicitly states the same thing.